src/model/resources/fusion/fusion.jl

```diff
@@ -0,0 +1,545 @@
+    @constraint(EP, [t in 1:T], vPulseUnderway[t] * component_size<=ecap)
+
+    # Core max uptime. If this parameter > 0,
+    # the fusion core must be cycled at least every n hours.
+    # Looks back over interior timesteps and ensures that a core cannot
+    # be committed unless it has been started at some point in
+    # the previous n timesteps
+    if has_max_pulse_length(reactor)
+        max_pulse_length = reactor.max_pulse_length
+        function starts_in_previous_hours(t)
+            vPulseStart[hoursbefore(p, t, 0:(max_pulse_length - 1))]
+        end
+        @constraint(EP, [t in 1:T], vPulseUnderway[t]<=sum(starts_in_previous_hours(t)))
+    end
+
+    if has_finite_starts(reactor)
+        max_starts = reactor.max_starts
+        @constraint(EP,
+            sum(vPulseStart[t] * ω[t] for t in 1:T) * component_size<=max_starts * ecap[y])
+    end
+end
+
+function fusion_pulse_status_linking_constraints!(
+        EP::Model,
+        inputs::Dict,
+        resource_component::AbstractString,
+        r_id::Int,
+        reactor::FusionReactorData,
+        vcommit::Symbol
+)
+    T = inputs["T"]
+
+    commit = EP[vcommit] # measured in number of components
+
+    from_model(f::Function) = EP[Symbol(f(resource_component))]
+
+    vPulseStart = from_model(fusion_pulse_start_name)
+    vPulseUnderway = from_model(fusion_pulse_underway_name)
+
+    # pulses cannot start unless the plant is committed
+    @constraint(EP, [t in 1:T], vPulseStart[t]<=commit[r_id, t])
+    @constraint(EP, [t in 1:T], vPulseUnderway[t]<=commit[r_id, t])
+end
+
+@doc raw"""
+    fusion_pulse_thermal_power_generation_constraint!(EP::Model,
+                              inputs::Dict,
+                              resource_component::AbstractString,
+                              r_id::Int,
+                              reactor::FusionReactorData,
+                              vp::Symbol,
+                              vcommit::Symbol)
+
+    Add constraint which acts on the power-output-like variable.
+"""
+function fusion_pulse_thermal_power_generation_constraint!(
+        EP::Model,
+        inputs::Dict,
+        resource_component::AbstractString,
+        r_id::Int,
+        reactor::FusionReactorData,
+        power_like::AbstractArray
+)
+    T = inputs["T"]
+
+    component_size = reactor.component_size
+    dwell_time = reactor.dwell_time
+
+    from_model(f::Function) = EP[Symbol(f(resource_component))]
+    ePulseStart = component_size * from_model(fusion_pulse_start_name)
+    ePulseUnderway = component_size * from_model(fusion_pulse_underway_name)
+
+    # Maximum thermal power generated by core y at hour y <= Max power of committed
+    # core minus power lost from down time at startup
+    @constraint(EP,
+        [t in 1:T],
+        power_like[r_id,
+            t]<=ePulseUnderway[t] -
+                _fusion_dwell_avoided_operation(dwell_time, ePulseStart[t]))
+end
+
+function fusion_max_fpy_per_year_constraint!(
```

Comment:
You could move this function after the following two functions to align with the workflow in `fusion_formulation_thermal_commit!`